src/ui/archive.js

```diff
@@ -27,10 +28,12 @@ function archiveUI (state) {
     title += `${keyEl(dat.key)}\n`
   }
   if (state.title) title += state.title
-  else if (state.writable) title += 'Sharing dat'
-  else title += 'Downloading dat'
+  else if (state.writable) {
+    title += 'Sharing dat'
+    if (dat.metadata) title += `: ${metadata(dat.metadata)}\nFiles`
+  } else title += 'Downloading dat'
   if (state.opts.sparse) title += `: ${state.opts.selectedFiles.length} ${pluralize('file', state.opts.selectedFiles.length)} (${pretty(state.selectedByteLength)})`
-  else if (stats.version > 0) title += `: ${stats.files} ${pluralize('file', stats.file)} (${pretty(stats.byteLength)})`
+  else if (stats.version > 0) title += `: ${stats.files} ${pluralize('file', stats.files)} (${pretty(stats.byteLength)})`
```

Comment:
Is this a bugfix unrelated to the PR?

Reply:
I felt that it was related! 
The bugfix _technically_ enhances the UI – in a roundabout way (haha), but I can remove it and submit a separate issue/pr! 😄

Reply:
Probably a good idea? 😉

Reply:
Probably! 
I'll rebase in the next while. 👌